### DISCUSSION TITLE
How to collect metrics from a Java application

### DISCUSSION THREAD

**Discussion by lizhiyoumai** [ccfos/nightingale#1325] [Q&A]
我想通过categraf采集Java应用的指标，然后上报到prometheus，包括我的一些自定义的指标。需要怎么做呢

**Comment by lizhiyoumai** [comment]
补充一下，prometheus是通过应用暴露端口的方式拉取指标数据的，categraf能做到类似的吗

**Comment by UlricQin** [comment]
应用暴露 `/metrics` 数据，categraf 使用 prometheus 来抓，配置文件在 `conf/input.prometheus/prometheus.toml`

**Reply by lizhiyoumai** [reply]
所以我理解categraf本身并不能直接采集应用指标数据，而是由prometheus来拉取应用指标数据，是吧？

**Reply by UlricQin** [reply]
`/metrics`接口只是统一遵从了prometheus的格式，并不一定非得prometheus来拉取，vmagent、categraf、telegraf、datadog-agent、grafana-agent，都可以拉。所以说 prometheus 来拉是不对的，就是各种 agent 来拉的

**Reply by lizhiyoumai** [reply]
如果我用categraf来拉取，应用的/metrics接口需要遵从什么样的标准呢，又是怎么去拉呢
